rust/src/collector.rs

```diff
@@ -60,7 +60,7 @@ impl Collector {
             }
             Event::Shutdown => {
                 let trace = TraceEvent {
-                    events: self.events.clone(),
+                    events: self.events.drain(..).collect(),
```

Comment:
@bhelx - 
This is the only real change in this PR... I ran `cargo fmt` on the code too which I should've done in another commit. Can you make sure you have a "format on save" kind of thing in your editor?